datalake/src/main/java/com/sequenceiq/datalake/service/sdx/dr/SdxBackupRestoreService.java

```diff
@@ -695,4 +698,47 @@ private static boolean isVersionEqual(SdxCluster cluster, String baseVersion) {
         Comparator<Versioned> versionComparator = new VersionComparator();
         return versionComparator.compare(cluster::getRuntime, () -> baseVersion) == 0;
     }
+
+    public boolean isDatalakeInBackupOrRestore(String sdxClusterName, String actorCrn) {
+        if (sdxClusterName == null || actorCrn == null) {
+            LOGGER.error("Didn't find datalake name or actor CRN. Can not validate if the datalake is in backup/restore. Continue to next step.");
+            return false;
+        }
+        DatalakeBackupStatusResponse backupStatus;
+        DatalakeRestoreStatusResponse restoreStatus;
+        try {
+            backupStatus = datalakeDrClient.getBackupStatus(sdxClusterName, actorCrn);
+            restoreStatus = datalakeDrClient.getRestoreStatus(sdxClusterName, actorCrn);
+        } catch (RuntimeException exception) {
+            Pattern pattern = Pattern.compile("Status information for backup|restore operation on datalake.*not found");
```

Comment:
I don't think we should rely on text pattern match to identify the validity of the flow. Can we do anything else? List backups/restore and check the latest from the list if any?

Reply:
Thank you @votez for checking this! I think we are actually using the same method to list all backup/restore and check the latest with the current code: getBackupStatus in CB will finally run -> backupDatalakeStatus in DatalakeDrService.java in TH -> this will run getLatestBackupStatus -> getDatalakeDrStatusEntitiesSorted or getDatalakeDrStatusEntityByDrOperationId. When we are running getLatestBackupStatus in TH it will throw the NOT_FOUND RuntimeException if they can not find a statusEntity (for example, if no backup has ever happened and stored in sdx_dr_status). I didn't find another way to check if this exception is the exception for the backup/restore not happened case though - do you have any other suggestions to check that? Based on what info we have in CB this is the way I can think of. Thank you and plz let me know what you think! =)